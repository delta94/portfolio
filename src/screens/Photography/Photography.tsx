import clsx from 'clsx';
import React from 'react';
import { createUseStyles } from 'react-jss';
import { useMediaQuery } from 'react-responsive';
import { Element } from 'react-scroll';
import { PhotographsContainer, VanishAnimation } from '../../components';
import { CustomTheme } from '../../theme';

interface PhotographyProps {}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  header: {
    color: theme.colors.white,
    marginLeft: theme.spacing(3),
    display: 'flex',
    justifyContent: 'center',
    cursor: 'pointer',
    marginBottom: theme.spacing(1.5),
  },
  photographyContainer: {
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
  },
  photographyContainerMobile: {
    flexWrap: 'wrap',
  },
  mainContainer: {
    marginTop: theme.spacing(1.5),
  },
}));

const Photography: React.FC<PhotographyProps> = () => {
  const classes: Record<
    'header' | 'photographyContainer' | 'mainContainer' | 'photographyContainerMobile',
    string
  > = useStyles();
  const isMobile: boolean = useMediaQuery({ maxWidth: 780 });

  return (
    <Element className={classes.mainContainer} name='Photography'>
      <div className={classes.header}>
        <VanishAnimation words={['Photography']} />
      </div>
      <div className={clsx([classes.photographyContainer, isMobile && classes.photographyContainerMobile])}>
        <PhotographsContainer />
      </div>
    </Element>
  );
};

export { Photography };
