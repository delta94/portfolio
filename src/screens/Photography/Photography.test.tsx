import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { Photography } from './Photography';

test('Photography renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <Photography />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
