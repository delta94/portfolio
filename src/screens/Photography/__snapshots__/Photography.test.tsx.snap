// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photography renders correctly 1`] = `
<DocumentFragment>
  <div
    name="Photography"
  >
    <div
      class="header-0-2-1"
    >
      <div
        style="opacity: 0; transform: translate3d(0,20px,0);"
      >
        <h2
          class="h2Text-0-2-4"
        >
          Photography
        </h2>
      </div>
    </div>
    <div
      class="photographyContainer-0-2-2"
    >
      <div
        class="mainContainer-0-2-5"
      >
        <div
          class="photosContainer-0-2-7"
        />
        <div
          class="button-0-2-8"
        >
          <button
            class="mainContainer-0-2-9"
            style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
          >
            Explore more images
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
